packages/backend/src/server/api/stream/channels/main.ts

```diff
@@ -11,8 +12,13 @@ export default class extends Channel {
 	public async init(params: any) {
 		// Subscribe main stream channel
 		this.subscriber.on(`mainStream:${this.user!.id}`, async data => {
+			let { body } = data;
+
 			switch (data.type) {
 				case 'notification': {
+					// Ignore notifications from instances the user has muted
+					if (isInstanceMuted(body, new Set<string>(this.userProfile?.mutedInstances ?? []))) return;
+
```

Comment:
```suggestion
			switch (data.type) {
				case 'notification': {
					// Ignore notifications from instances the user has muted
					if (isInstanceMuted(data.body, new Set<string>(this.userProfile?.mutedInstances ?? []))) return;
```

TypeScriptの仕様上、Destructuring assignmentするとswitch-case内でtypeがおかしくなってしまうので出来れば避けてください。:pray:

Reply:
Added your changes, thank you!